src/equations/lattice_boltzmann_2d.jl

```diff
@@ -0,0 +1,460 @@
+                                             surface_flux_function, equation)
+end
+
+
+# Pre-defined source terms should be implemented as
+# function source_terms_WHATEVER(u, x, t, equation::LatticeBoltzmannEquation2D)
+
+
+# Calculate 1D flux in for a single point
+@inline function calcflux(u, orientation, equation::LatticeBoltzmannEquation2D)
+  if orientation == 1
+    v_alpha = equation.v_alpha1
+  else
+    v_alpha = equation.v_alpha2
+  end
+  return v_alpha .* u
+end
+
+
+function flux_lax_friedrichs(u_ll, u_rr, orientation, equation::LatticeBoltzmannEquation2D)
+  if orientation == 1
+    v_alpha = equation.v_alpha1
+  else
+    v_alpha = equation.v_alpha2
+  end
+  return 0.5 * ( v_alpha .* (u_ll + u_rr) - abs.(v_alpha) .* (u_rr - u_ll) )
+end
+
+
+"""
+    density(u, equation::LatticeBoltzmannEquation2D)
+
+Calculate the macroscopic density from the particle distribution functions `u`.
+"""
+density(u, equation::LatticeBoltzmannEquation2D) = sum(u)
+
+
+"""
+    velocity(u, orientation, equation::LatticeBoltzmannEquation2D)
+
+Calculate the macroscopic velocity for the given `orientation` (1 -> x, 2 -> y) from the
+particle distribution functions `u`.
+"""
+function velocity(u, orientation, equation::LatticeBoltzmannEquation2D)
+  if orientation == 1
+    v_alpha = equation.v_alpha1
+  else
+    v_alpha = equation.v_alpha2
+  end
+  
+  return sum(v_alpha .* u)/density(u, equation)
+end
+
+
+"""
+    velocity(u, equation::LatticeBoltzmannEquation2D)
+
+Calculate the macroscopic velocity vector from the particle distribution functions `u`.
+"""
+function velocity(u, equation::LatticeBoltzmannEquation2D)
+  @unpack v_alpha1, v_alpha2 = equation
+  rho = density(u, equation)
+  
+  return SVector(sum(v_alpha1 .* u)/rho, sum(v_alpha2 .* u)/rho)
+end
+
+
+"""
+    pressure(u, equation::LatticeBoltzmannEquation2D)
+
+Calculate the macroscopic pressure from the particle distribution functions `u`.
+"""
+pressure(u, equation::LatticeBoltzmannEquation2D) = density(u, equation) * equation.c^2 / 3
+
+
+"""
+    equilibrium_distribution(alpha, rho, v1, v2, equation::LatticeBoltzmannEquation2D)
+
+Calculate the local equilibrium distribution for the distribution function with index `alpha` and
+given the macroscopic state defined by `rho`, `v1`, `v2`.
+"""
+function equilibrium_distribution(alpha, rho, v1, v2, equation::LatticeBoltzmannEquation2D)
+  @unpack weights, c_s, v_alpha1, v_alpha2 = equation
+
+  va_v = v_alpha1[alpha]*v1 + v_alpha2[alpha]*v2
+  cs_squared = c_s^2
+  v_squared = v1^2 + v2^2
+
+  return weights[alpha] * rho * (1 + va_v/cs_squared
+                                   + va_v^2/(2*cs_squared^2)
+                                   - v_squared/(2*cs_squared))
+end
+
+
+function equilibrium_distribution(alpha, u, equation::LatticeBoltzmannEquation2D)
+  rho = density(u, equation)
+  v1, v2 = velocity(u, equation)
+
+  return equilibrium_distribution(alpha, rho, v1, v2, equation)
+end
```

Comment:
Is this used anywhere? If not, we might remove it.

Reply:
Removed in 4442617b.